Add release guide (#124)
apache/incubator-seatunnel-website: community/contribution_guide/release.md

     <n>m = key expires in n months
     <n>y = key expires in n years
Key is valid for? (0)
Key does not expire at all
Is this correct? (y/N) y

GnuPG needs to construct a user ID to identify your key.

Real name: ${Input username}
Email address: ${Input email}
Comment: ${Input comment}
You selected this USER-ID:
   "${Inputed username} (${Inputed comment}) <${Inputed email}>"

Change (N)ame, (C)omment, (E)mail or (O)kay/(Q)uit? O
You need a Passphrase to protect your secret key. # Input passwords
```

### Check Generated Key

```shell
gpg --list-keys
```

Execution Result:

```shell
pub   4096R/85E11560 2019-11-15
uid                  ${Username} (${Comment}) <{Email}>
sub   4096R/A63BC462 2019-11-15
```

Among them, 85E11560 is public key ID.

### Upload the Public Key to Key Server

The command is as follow:

```shell
gpg --keyserver pgpkeys.mit.edu/ --send-key 85E11560
```
* Here pgpkeys.mit.edu is a random selection of keyserver. Any key server from the list https://sks-keyservers.net/status/ is acceptable because they are automatically synchronized.

`pool.sks-keyservers.net` is randomly chosen from [public key server](https://sks-keyservers.net/status/).

> Each server will automatically synchronize with one another, so it would be okay to choose any one.

Reviewer: Can you add the description of " Search Key in key server" to verify whether your key can be found?

```shell
gpg --keyserver ${key server} --search-keys {public key ID}
```

## Apache Maven Central Repository Release

### Set settings.xml

Add the following template to `~/.m2/settings.xml`, all the passwords need to be filled in after encryption.
For encryption settings, please see [here](http://maven.apache.org/guides/mini/guide-encryption.html).

```xml
<settings>
  <servers>
    <server>
      <id>apache.snapshots.https</id>
      <username> <!-- APACHE LDAP username --> </username>
      <password> <!-- APACHE LDAP encrypted password --> </password>
    </server>
    <server>
      <id>apache.releases.https</id>
      <username> <!-- APACHE LDAP username --> </username>
      <password> <!-- APACHE LDAP encrypted password --> </password>
    </server>
  </servers>
</settings>
```

### Update Release Notes

```

> https://github.com/apache/incubator-seatunnel/blob/dev/ReleaseNotes.md

Reviewer: ```suggestion
https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md
```

```

### Create Release Branch

Suppose SeaTunnel source codes downloaded from GitHub is under `~/incubator-seatunnel/` directory and the version to be released is `${RELEASE.VERSION}`.
Create `${RELEASE.VERSION}-release` branch, where all the following operations are performed.

```shell
cd ~/incubator-seatunnel/
git pull
git checkout -b ${RELEASE.VERSION}-release
git push origin ${RELEASE.VERSION}-release
```

### Pre-Release Check

```shell

> mvn release:prepare -Prelease -Darguments="-DskipTests" -DdryRun=true -Drevison=${RELEASE-VERSION} -Dusername=${Github username}

Reviewer: ```suggestion
mvn release:prepare -Drelease=true -Darguments="-DskipTests" -DdryRun=true -Drevison=${RELEASE-VERSION} -Dusername=${Github username}
```
https://github.com/apache/incubator-seatunnel/pull/2703

```

> -Prelease: choose release profile, which will pack all the source codes, jar files and executable binary packages.

Reviewer: ```suggestion
-Drelease=true: choose release profiles, which will pack all the source codes, jar files and executable binary packages.
```

> --Drevison=${RELEASE-VERSION}：it can make the version number is inputted.

Reviewer: ```suggestion
--Drevision=${RELEASE-VERSION}：it can make the version number is inputted.
```

-DdryRun=true：rehearsal, which means not to generate or submit new version number and new tag.

### Prepare for the Release

First, clean local pre-release check information.

```shell
mvn release:clean
```

Then, prepare to execute the release.

```shell

> mvn release:prepare -Prelease -Darguments="-DskipTests"  -Drevison=${RELEASE-VERSION} -DpushChanges=false -Dusername=${Github username}

Reviewer: ```suggestion
mvn release:prepare -Drelease=true -Darguments="-DskipTests"  -Drevision=${RELEASE-VERSION} -DpushChanges=false -Dusername=${Github username}
```

```

It is basically the same as the previous rehearsal command, but deleting -DdryRun=true parameter.

-DpushChanges=false：do not submit the edited version number and tag to Github automatically.

After making sure there is no mistake in local files, submit them to GitHub.

```shell
git push
git push origin --tags
```

### Deploy the Release

```shell

> mvn release:perform -Prelease -Darguments="-DskipTests" -Drevison=${RELEASE-VERSION} -Dusername=${Github username}

Reviewer: ```suggestion
mvn release:perform -Drelease=true -Darguments="-DskipTests" -Drevision=${RELEASE-VERSION} -Dusername=${Github username}
```

```
## Check Release

### Check sha512 hash

```shell
shasum -c apache-seatunnel-incubating-${RELEASE.VERSION}-src.tar.gz.sha512
shasum -c apache-seatunnel-incubating-${RELEASE.VERSION}-bin.tar.gz.sha512
```

### Check gpg Signature

First, import releaser's public key.
Import KEYS from SVN repository to local. (The releaser does not need to import again; the checking assistant needs to import it, with the user name filled as the releaser's. )

```shell
curl https://dist.apache.org/repos/dist/dev/incubator/seatunnel/KEYS >> KEYS
gpg --import KEYS
gpg --edit-key "${GPG username of releaser}"
  > trust

Please decide how far you trust this user to correctly verify other users' keys
(by looking at passports, checking fingerprints from different sources, etc.)

  1 = I don't know or won't say
  2 = I do NOT trust
  3 = I trust marginally
  4 = I trust fully
  5 = I trust ultimately
  m = back to the main menu

Your decision? 5

  > save
```

Then, check the gpg signature.

```shell
gpg --verify apache-seatunnel-incubating-${RELEASE.VERSION}-src.tar.gz.asc apache-seatunnel-incubating-${RELEASE.VERSION}-src.tar.gz

> gpg --verify apache-seatunnel-incubating-${RELEASE.VERSION}-seatunnel-bin.tar.gz.asc apache-seatunnel-incubating-${RELEASE.VERSION}-seatunnel-bin.tar.gz

Reviewer 2: ```suggestion
gpg --verify apache-seatunnel-incubating-${RELEASE.VERSION}-bin.tar.gz.asc apache-seatunnel-incubating-${RELEASE.VERSION}-seatunnel-bin.tar.gz
```

Reviewer: ```suggestion
gpg --verify apache-seatunnel-incubating-${RELEASE.VERSION}-bin.tar.gz.asc apache-seatunnel-incubating-${RELEASE.VERSION}-bin.tar.gz
```
